### ISSUE TITLE
Can't filter dynamically created content

### ISSUE THREAD

**Issue by awl19** [no81no/jplist#117]
Just want to say first off, jPList is an amazing plugin and hats off to the author of it.

On to my issue. I'm pulling about 200 "items" (rows in a database) and placing them into jPList. That works great, but the problem is for each of those items, there's a column for categories. That column contains a comma separated list of ID's being pulled from another table that holds the categories. Anyways, I had originally written the code to pull the categories on the same page as the "items", but it proved to be too much of a drain on the web server's load. So instead I'm using JQuery to load the categories for each "item".
```javascript
            $('.tr').each(function(index){
                var $this = $(this);
                var $vendor_id = $($this).find('input[name="vendor_id"]');
                var $category = $($this).find('.category');
                $($this).delay(400*index,function() {
                    $.post(
                        "<?php echo URLADDR ?>functions/category-get.func.php",
                        $($vendor_id).serialize(),
                        function(data) {
                            $($category).find('.loader').fadeOut(500);
                            $($category).html(data);
                            $($category).find('a').fadeIn(500).css('display','inline-block');
                        }
                    );
                });
            });
```
Here's an example of the HTML for the jPList dropdown:
```
<span data-path="auto-accessories">Auto Accessories</span>
```
Here's an example of the HTML being pulled from the JQuery post above:
```
<a style="display: inline-block;" class="btn btn-dark auto-accessories" href="#">Auto Accessories</a>
```

When I select a category to filter the list shows no results, as though that category doesn't exist. But the problem didn't occur when I wasn't using JQuery to post the categories.

Any help would be greatly appreciated.

**Comment by no81no** [comment]
There are 2 options for this issue.
The first one is to load all your items, and only after that to call jPList plugin.
It can be achieved using promises:
https://api.jquery.com/category/deferred-object/
Or just with flags.

The second option could be to use jPList API to add your items dynamically once they are loaded with ajax. Please check API here:
https://jplist.com/documentation/api

**Comment by awl19** [comment]
Thank you so much! This is exactly what I was working for =)

**Comment by no81no** [comment]
You're welcome :)

**Comment by awl19** [comment]
Hey, sorry to re-open this thread, but it appears I'm too dim to make this work XD

This is what I've got:
```javascript
            var $jplist = $('#jplist');
            $jplist.jplist({
                itemsBox: '.vendors'
                , itemPath: '.tr'
                , panelPath: '.jplist-panel'
                //save plugin state
                , storage: 'localstorage'
                , storageName: 'jplist-div-layout'
                , effect: 'fade'

            });
            $('.tr').each(function(index){
                var $this = $(this);
                var $id = $($this).attr('id');
                $id = '#' + $id;
                var $vendor_id = $($this).find('input');
                $($this).delay(400*index,function() {
                    $.post(
                        "<?php echo URLADDR ?>functions/vendors-get.func.php",
                        $($vendor_id).serialize(),
                        function(data) {
                            $($this).html(data, function() {
                                $jplist.jplist({
                                    command: 'add'
                                    ,commandData: {
                                        $items: $($id)
                                    }
                                });
                            });
                        }
                    );
                });
            });
```
So, here's the line where the HTML is added to the page:
```javascript
$($this).html(data, function() {
```
And within that (because I want to wait until the HTML is created) is the jPList code to add the item:
```javascript
                                $jplist.jplist({
                                    command: 'add'
                                    ,commandData: {
                                        $items: $($id)
                                    }
                                });
```
I'm pulling the ID from each div and telling jPList to add that item. I tested $id with an alert to make sure it was coming out properly and it was.

Now, the HTML populates, as expected, but still jPList is not registering the items as though they exist. So, I'm not sure where I'm going wrong.

**Comment by no81no** [comment]
Please try the following code:

```javascript
//ajax requests counter
var counter = 0;

$('.tr').each(function(index){
	var $this = $(this);
	var $vendor_id = $($this).find('input[name="vendor_id"]');
	var $category = $($this).find('.category');
	$($this).delay(400*index,function() {
		$.post(
			"<?php echo URLADDR ?>functions/category-get.func.php",
			$($vendor_id).serialize(),
			function(data) {
				$($category).find('.loader').fadeOut(500);
				$($category).html(data);
				$($category).find('a').fadeIn(500).css('display','inline-block');
				
				//for every successful ajax request -> +1
				counter++;
				
				//if all ajax request were successful
				if(counter >= $('.tr').length){
				
					//start jplist
					$jplist.jplist({
						itemsBox: '.vendors'
						, itemPath: '.tr'
						, panelPath: '.jplist-panel'
						//save plugin state
						, storage: 'localstorage'
						, storageName: 'jplist-div-layout'
						, effect: 'fade'
					});
				}
			}
		);
	});
});
```

**Comment by awl19** [comment]
Hahaha, wow. I didn't think something like that would work. You're the best! Thank you =)

**Comment by no81no** [comment]
You are welcome :)
